.github/workflows/cross-arch-build-test.yml

```diff
@@ -0,0 +1,314 @@
+            binary: kubectl-oadp-windows-amd64.exe
+          - os: windows-11-arm  # arm64
+            arch: arm64
+            binary: kubectl-oadp-windows-arm64.exe
+    
+    runs-on: ${{ matrix.os }}
+    
+    steps:
+    - name: Checkout code (for tests)
+      uses: actions/checkout@v4
+
+    - name: Set up Go (for running tests)
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+        cache: true
+
+    - name: Download Windows binaries
+      uses: actions/download-artifact@v4
+      with:
+        name: windows-binaries
+
+    - name: Verify runner architecture matches binary
+      run: |
+        echo "Expected architecture: ${{ matrix.arch }}"
+        echo "Runner OS: ${{ matrix.os }}"
+        echo "Testing binary: ${{ matrix.binary }}"
+        if (Test-Path ${{ matrix.binary }}) {
+          echo "✅ Binary found: ${{ matrix.binary }}"
+        } else {
+          echo "❌ Binary not found: ${{ matrix.binary }}"
+          exit 1
+        }
+
+    - name: Test binary execution (native)
+      run: |
+        echo "Testing ${{ matrix.binary }} on native ${{ matrix.arch }} runner..."
+        .\${{ matrix.binary }} --help
+        echo "✅ Binary execution successful on ${{ matrix.arch }}"
+
+    - name: Install and test as kubectl plugin
+      run: |
+        mkdir -p C:\tmp\bin
+        copy ${{ matrix.binary }} C:\tmp\bin\kubectl-oadp.exe
+        $env:PATH = "C:\tmp\bin;" + $env:PATH
+        
+        echo "Testing kubectl plugin functionality on ${{ matrix.arch }}..."
+        kubectl-oadp --help
+        kubectl-oadp version --help  
+        kubectl-oadp nonadmin --help
+        echo "✅ kubectl plugin tests passed on ${{ matrix.arch }}"
+
+    - name: Run test suite
+      run: |
+        echo "Running full test suite..."
+        make test
+        echo "✅ Test suite passed"
+
+  # Summary
+  test-summary:
+    name: Test Summary
+    runs-on: ubuntu-latest
+    needs: [build-all, test-linux, test-macos, test-windows]
+    if: always()
+    
+    steps:
+    - name: Check all results
+      run: |
+        echo "=== Cross-Architecture Build & Test Summary ==="
+        echo ""
+        
+        if [ "${{ needs.build-all.result }}" = "success" ]; then
+          echo "✅ Multi-arch build: PASSED"
+        else
+          echo "❌ Multi-arch build: FAILED"
+        fi
+        
+        if [ "${{ needs.test-linux.result }}" = "success" ]; then
+          echo "✅ Linux tests (amd64 + arm64): PASSED"
+        else
+          echo "❌ Linux tests: FAILED"
+        fi
+        
+        if [ "${{ needs.test-macos.result }}" = "success" ]; then
+          echo "✅ macOS tests (Intel + Apple Silicon): PASSED"
+        else
+          echo "❌ macOS tests: FAILED"
+        fi
+        
+        if [ "${{ needs.test-windows.result }}" = "success" ]; then
+          echo "✅ Windows tests (amd64 + arm64 native): PASSED"
+        else
+          echo "❌ Windows tests: FAILED"
+        fi
```

Comment:
Good use of `&&` operators here! This ensures that the success message only prints if ALL the previous jobs succeeded. Without these operators, the condition would only check the first job's status.